plone/app/contenttypes/browser/collection.py

```diff
@@ -13,8 +13,16 @@ class CollectionView(FolderView):
     def __init__(self, *args, **kwargs):
         super(CollectionView, self).__init__(*args, **kwargs)
         context = aq_inner(self.context)
-        self.collection_behavior = ICollection(context)
-        self.b_size = self.collection_behavior.item_count
+
+        try:
+            # the object may have the behavior applied
+            self.behavior = ICollection(context)
```

Comment:
yes, I had this doubt; currently we are using 2 different names: `self.collection_behavior` and `wrapped`; I can leave it as self.`collection_behavior` or use `self.wrapped`; let me know.